sympy/ntheory/generate.py

```diff
@@ -55,11 +55,28 @@ class Sieve:
     """
 
     # data shared (and updated) by all Sieve instances
-    def __init__(self):
+    def __init__(self, sieve_interval=1_000_000):
+        """ Initial parameters for the Sieve class.
+
+        Parameters
+        ==========
+
+        sieve_interval (int): Amount of memory to be used
+
+        Raises
+        ======
+
+        ValueError
+            If ``sieve_interval`` is not positive.
+
+        """
         self._n = 6
-        self._list = _aset(2, 3, 5, 7, 11, 13) # primes
+        self._list = _array('L', [2, 3, 5, 7, 11, 13]) # primes
         self._tlist = _aset(0, 1, 1, 2, 2, 4) # totient
         self._mlist = _aset(0, 1, -1, -1, 0, -1) # mobius
```

Comment:
tlist could be array type "L" and mlist array type "i". Then `_aset` could be removed and `_arange` changed to use type "L"

Reply:
Remove `_azeros`, `_aset`, and `_arange`, since it is no longer worthwhile to have them as functions.